### ISSUE TITLE
 template {{Label}} does not work

### ISSUE THREAD

**Issue by josefkreitmayer** [TransforMap/base.transformap.co#1]
The template {{Label}} does not work.
 ---

The bug relates to the **inability to create human readable labels in the places of the q-numbers**, when generating wiki-pages to describe a single taxonomy. The template {label} does not work, and just the Q-Numbers are visible. The labels should be displayed in the UI language.

**You can try it out:** Try to produce a wiki-page that contains a semantically linked item. It can be any item from the all items list: https://base.transformap.co/wiki/Special:AllPages?from=&to=&namespace=120

**Example:** 

Currently any kind of taxonomy-item in wikibase is presented in a general overview ordered as added.
Currently there is not way to produce overview pages with the distinct taxonomies, due to the error described. @species tried to put all items of one taxonomy on one page, https://base.transformap.co/wiki/SSEDAS_Taxonomy to have that as viewpoint for the contributors to that specific taxonomy. We want to do so for every taxonomy created.

Assumptions by @species 
- can it be that some sort of index has to be generated?
-  can it be that we are am using the "Item:"-namespace for entities and not the main namespace like Wikidata?

**Comment by gandhiano** [comment]
Please help me understand this problem further, by clarifying the concepts that we are playing with.

We have [items in the Wikibase](https://base.transformap.co/wiki/Special:AllPages?from=&to=&namespace=120), which can basically be seen as tags that can be assigned to points. Here you can add items such as Foodbank, Guerrilla Gardening, but also Transformap or SSEDAS Taxonomy.

Then we have [properties](https://base.transformap.co/wiki/Special:ListProperties), which make sense of the relations between different items, in order to build up structures for them. Here we have, for example, that a Foodbank is an instance of "Type of initiative" and a subclass of "Social inclusion" (and also a "Type of Initiative Tag foodbank, which I don't understand, but that's probably another issue). We also have that SSEDAS Taxonomy is an instance of "Taxonomy".

Therefore, I don't understand what is missing and what you want to achieve with having yet another tag? 

Also, please clarify if we are still following the logic, that the items and properties are entered as commons, with each contribution available to everyone, and that the "taxonomy" itself is built by the specific users/use cases on the query level? Here the [related reading](https://discourse.transformap.co/t/leave-it-to-the-user-collectively-shaping-taxonomies-to-express-plurality-of-narratives/1030) on the discussions and proposals we had around this.

**Comment by josefkreitmayer** [comment]
I give an overview of the current setup from the user perspective from a user story perspective:

- I (Persona Angelo) want to produce a map for the international community of alternative nutrition and want to get all existing data out of the transformap system on a map, that is relating to food. I call my map and filter system the "ALTFOOD" taxonomy. I want to use current filters, and develop new ones. In most of the current filters I am not interested, just the food ones. 

On the ["list all items"](https://base.transformap.co/wiki/Special:AllPages?from=&to=&namespace=120) page, I find an overview of alllllll the items, that are in the taxonomy server, but I am just interested to get an overview on the food ones, to discuss them with my peers, and develop new ones.

**How can I produce an overview of items, that are of my interest, and lastly an overview of just the "ALTFOOD" taxonomy, that I want to produce? I do not care about technical specifications, and why what is in which place, I just want to work with it, and am willing to stick to the established relations. Important for me is, that I do not hard-copy any taxonomy-segments in my overview-page, as I also want the translations to go with it. Our project is international with many languages.**

**Comment by josefkreitmayer** [comment]
In much shorter terms:

- the problem is just technical, as the labels of items are not produced in any pages apart from the system-generated pages.
- the problem is not related to the properties, but as the properties are also items (P-numbers instead of q-numbers), the problem might also apply to them. You can try it out: Try to produce a wiki-page that contains a semantically linked item. https://base.transformap.co/wiki/Special:ListProperties
- of course the idea is still 

>"anyone - from collective efforts of networks or organisations, to single individuals - and, to the extent possible, independently of the technical affinity, can shape the information that is displayed, by working on filter sets (basically an aggregate selection of criteria of properties the filtered objects such have, such as e.g. specific categories, tags, sources of data, etc.)"

Actually that is the very reason why to work on this bug!

Otherwise it is just noise, once more than 2 taxonomies (or to say perspectives / sets of how to display information on a map) are created.

**Comment by josefkreitmayer** [comment]
@gandhiano I notice, also for myself, that a guide for creating taxonomies with wikibase is needed to work with it: https://tree.taiga.io/project/transformap/us/399 "create a compelling guide for creating taxonomies with Wikibase"

**Comment by species** [comment]
I've tried to get an answer on the #wikidata IRC channel, but on several requests no one seem to be able to answer that.

> Can anyone here help me with the Label-Template on our custom Wikibase install? It does not output the translations as it does on Wikidata... can it be that some sort of index has to be generated on Wikibase-client?

**Comment by species** [comment]
We need someone that *really* knows how Mediawiki templates work, in particular with LUA and the Wikibase extension.

**Comment by gandhiano** [comment]
From your description I understand that this is not affecting the usage for a mapping workshop. The attributes (items, properties) will all be there and it is a matter of generating separated views/templates according to context, *afterwards*.

**Comment by josefkreitmayer** [comment]
@gandhiano, you are right, it does not interfere with the core functionality. It just get´s messy, especially, as Susy-map people are still translating their items, and will then be presented with new items unfamiliar to them. Additional to that problem, it is very difficult to work with the translation and distinction of serveral taxonomies at once without generating that overview pages for each distinct taxonomy.

**Comment by almereyda** [comment]
Indeed I see two issues being discussed here:

1. Having an overview of items referencing a certain Taxonomy, as in
  * https://base.transformap.co/wiki/Item:Q8
    * https://base.transformap.co/wiki/Special:WhatLinksHere/Item:Q8
  * https://base.transformap.co/wiki/Item:Q4
    * https://base.transformap.co/wiki/Special:WhatLinksHere/Item:Q4
2. Making the `{{Label|Q8}}` tag work on https://base.transformap.co/wiki/SSEDAS_Taxonomy

Am I correct? Then we need individual issues.

**Comment by almereyda** [comment]
I have asked Lucas from Wikimedia and he pointed me to the configuration of [the Scribunto plugin](https://www.mediawiki.org/wiki/Extension:Scribunto), which necessarily needs a LUA interpreter provided along with it.

* [ ] Check for availability of a LUA interpreter within the Wikibase container.
* [ ] Check for configuration options of the Scribunto plugin in `LocalSettings.php`
